pipelines/batch/src/main/java/com/google/fhir/analytics/EtlUtils.java

```diff
@@ -86,21 +86,41 @@ static List<PipelineResult> runMultiplePipelinesWithTimestamp(
   static List<PipelineResult> runMultipleMergerPipelinesWithTimestamp(
       List<Pipeline> pipelines, ParquetMergerOptions options, FhirContext fhirContext)
       throws IOException {
-    Instant instant1 =
-        DwhFiles.forRoot(options.getDwh1(), fhirContext)
-            .readTimestampFile(DwhFiles.TIMESTAMP_FILE_START);
-    Instant instant2 =
-        DwhFiles.forRoot(options.getDwh2(), fhirContext)
-            .readTimestampFile(DwhFiles.TIMESTAMP_FILE_START);
-    Instant mergedInstant = (instant1.compareTo(instant2) > 0) ? instant1 : instant2;
-    DwhFiles.forRoot(options.getMergedDwh(), fhirContext)
-        .writeTimestampFile(mergedInstant, DwhFiles.TIMESTAMP_FILE_START);
+    mergeWithLatestTimestamp(
+        options.getDwh1(),
+        options.getDwh2(),
+        options.getMergedDwh(),
+        DwhFiles.TIMESTAMP_FILE_START,
+        fhirContext);
+    // Transaction file exists for Batch Export mode, merge the timestamp file in this case
+    if (DwhFiles.forRoot(options.getDwh1(), fhirContext)
```

Comment:
How do we handle the edge case where someone has a DWH created using non-bulk-export methods and then they switch to bulk-export in their config? I don't think we are handling that case properly right now, so we should either throw an error/warning (and do nothing) or handle it gracefully.